heimlig/src/crypto/ed25519.rs

```diff
@@ -0,0 +1,219 @@
+use crate::crypto::Error;
+use ed25519_dalek::{SecretKey, Signature, Signer, SigningKey, Verifier, VerifyingKey};
+
+/// Ed25519 signature size in bytes.
+pub const SIGNATURE_SIZE: usize = ed25519_dalek::SIGNATURE_LENGTH;
+/// Ed25519 private key size in bytes.
+pub const PRIVATE_KEY_SIZE: usize = ed25519_dalek::SECRET_KEY_LENGTH;
+/// Ed25519 public key size in bytes.
+pub const PUBLIC_KEY_SIZE: usize = ed25519_dalek::PUBLIC_KEY_LENGTH;
+
+/// Signs the message using the Ed25519 algorithm.
+///
+/// # Arguments
+///
+/// * `private_key`: A slice containing private key bytes.
+///   The private key has to be `PRIVATE_KEY_SIZE` bytes long.
+/// * `message`: A slice containing the message to sign bytes.
+/// * `signature`: A mutable slice where the computed signature will be stored.
+///   The signature slice length has to be `SIGNATURE_SIZE` bytes long.
+///
+/// # Errors
+///
+/// The function returns an error if:
+/// * `InvalidBufferSize`: The length of the `signature` is not `SIGNATURE_SIZE` bytes.
+/// * `InvalidPrivateKey`: The length of the `private_key` is not `PRIVATE_KEY_SIZE` bytes.
+pub fn ed25519_sign(private_key: &[u8], message: &[u8], signature: &mut [u8]) -> Result<(), Error> {
+    if signature.len() != SIGNATURE_SIZE {
```

Comment:
We should think about also allowing output buffers that are larger than required.

Reply:
It will be done for all algorithms in a separate commit.